README.md

```diff
@@ -29,23 +29,145 @@ This proposal reached Stage 1 at the 2020 Feb TC39 meeting.
+  * Nano Second
+
+#### Design
+We are going to support the following fields as a beginning
+  * Week
+  * Day
+  * Hour
+  * Minute
+  * Second
+  * Millisecond
+  * Microsecond
+  * Nano Second
+
+### Determining Time Fields
+Users need to determine the time fields,
+ * For example: 6059 seconds
+   * Hours, Minutes, Seconds
+     * 1 hours, 40 minutes and  59 seconds
+   * Minutes and Seconds
+     * 100 minutes and 59 seconds.
+ 
+#### Design
+   * Users can determine the time fields using `LargestField` and `SmallestField` parameters.
+   * Example: 
+   ```javascript
+    {
+      LargestField: "hour",
+      SmallestField: "second"
+    }
+  ```
+   * Default values
+    ```javascript
+         LargestField: "Day",
+         SmallestField: "second"
+    ```
+### Hide zero-value fields
+Users needs to hide zero-value fields depends in some criteria
+  * Hide all zero-value fields
+    * Hours, Minutes, Seconds (without the `hide` option)
+      * 1 hours, 0 minutes and 59 seconds.
+    * Hours, Minutes, Seconds (with the `hide` option)
+      * 1 hours and 59 seconds.
+  * Hide-upper-limit with zero-valued.
+    * Hours, Minutes, Seconds, MilliSeconds (without the `hide-upper-limit` option)
+      * 0 hr, 20 min, 0 sec, 100 msec .
+    * Hours, Minutes, Seconds MilliSeconds(with the `hide-upper-limit` option)
+      * 20 min, 0 sec, 100 msec.
+  * Hide-lower-limit with zero-valued.
+
+#### Design
+- Users can set the `HidingZeroValuedOption` parameter  by one of the following values:
+    * “ALL”, // Hide all the fields that have a zero-valued.
+    * “LEADING_AND_TRAILING”, // Hide all the zero fields in the leading or the training
+    * “LEADING_ONLY”,
+    * “TRAILING_ONLY”, 
+    * “NONE”, // Do not hide any zero-valued fields.
+
+- Default Value: “NONE”
+
+### Locale aware format
+ - Users needs the formatting to be dependent on the locale
+ - For example:
+    * en-US
+      * 1 hour, 46 minutes and 40 seconds
+    * fr-FR
+      * 1 heure, 46 minutes et 40 secondes
 
 ## API design
+I propose to have a class called `DurationFormat` 
 
-  ```javascript
-    const formatter =  new Intl.DurationFormatter('en-US' , { style: 'short', fields: 'hours-minutes-seconds'});
 
-    const duration =  Temporal.Duration.from({
-                        hours: 2, minutes: 46, seconds: 40
-                      });
+  ```javascript
+    const formatter =        
+        new Intl.DurationFormat({
+                              locale: 'en-US',
+		                		      largestField: 'hour',
+  					                  smallestField: 'second', 
+					                    style: 'short',
+					                    round: true,
+					                    HidingZeroValuedOption: NONE,
+                              });
+
+    const duration = 
+        Temporal.Duration.from({
+            hours: 2, minutes: 46, seconds: 40
+         });
 
     console.log(formatter.format(duration));
-    // output: 2 hr 46 min 40 sec
+      // output: 2 hr 46 min 40 sec
```

Comment:
I propose that we move this example to somewhere close to the top of this document for clarity.